src/systems/user_commands/UserCommands.cc

```diff
@@ -414,9 +414,31 @@ bool CreateCommand::Execute()
     }
     case msgs::EntityFactory::kLight:
     {
-      // TODO(louise) Support light msg
-      ignerr << "light field not yet supported." << std::endl;
-      return false;
+      auto lightMsg = createMsg->light();
+      // Check if there's already a top-level entity with the given name
+      if (kNullEntity != this->iface->ecm->EntityByComponents(
+          components::Name(lightMsg.name()),
+          components::ParentEntity(lightMsg.parent_id())))
+      {
+        ignwarn << "Light entity named [" << lightMsg.name() << "] already"
+                << " exists. Entity not spawned."
+                << std::endl;
+        return false;
+      }
+
+      sdf::Light lightSDF = convert<sdf::Light>(lightMsg);
```

Comment:
How about just populating `root` with `lightSDF` and letting the rest of this function handle things like naming and pose? You'll probably need to set `isLight` to true. This way there's less duplication and fields like `allow_renaming` are also used.